app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/CharacterServicesImpl.kt

```diff
@@ -8,6 +8,18 @@ import io.reactivex.Observable
 
 class CharacterServicesImpl(private val api: MarvelResquestGenerator = MarvelResquestGenerator(), private val mapper: CharacterMapperService = CharacterMapperService()) {
 
+    fun getCharacterById(id: Int?): Observable<Character> {
```

Comment:
id shouldn't be null. And please avoid null asertion (!!)

Reply:
done